### PR TITLE
Follow up for KIP-103 changes that fixes ProducerPool's extraction of endpoints to use the security protocol in the URL instead of the listener name.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@ijuma @theduderog Same deal as confluentinc/schema-registry#474. In this case there isn't an explicit option for setting bootstrap servers because you can choose to override any producer configs, so not sure where we'd document the limitation that you need to explicitly set the bootstrap servers if you have a cluster where only some endpoints will be accessible to REST proxy. In SimpleConsumer the previous build fix was ok since we're filtering to PLAINTEXT and only passing host/port to the SimpleConsumer.

File: src/main/java/io/confluent/kafkarest/ProducerPool.java [hunk 1]
```diff
@@ -143,7 +144,8 @@ private static String getBootstrapBrokers(ZkUtils zkUtils) {
         if (bootstrapBrokers.length() > 0) {
           bootstrapBrokers += ",";
         }
-        bootstrapBrokers += ep.connectionString();
+        String hostport = ep.host() == null ? ":" + ep.port() : Utils.formatAddress(ep.host(), ep.port());
+        bootstrapBrokers += ep.securityProtocol() + "://" + hostport;
```
Comment (theduderog) [inline src/main/java/io/confluent/kafkarest/ProducerPool.java]: this is legit for the bootstrap servers config param?

Comment (theduderog) [inline src/main/java/io/confluent/kafkarest/ProducerPool.java]: NM.  Found it:  https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L52-L54

Question:  any particular reason to add the protocol prefix?

Comment (theduderog) [inline src/main/java/io/confluent/kafkarest/ProducerPool.java]: Ooops.  Looking more closely, it accepts backslashes but not forward slashes.  Go figure.

Comment (ijuma) [inline src/main/java/io/confluent/kafkarest/ProducerPool.java]: It works with and without as discussed in https://github.com/confluentinc/schema-registry/pull/474/files/2c3cf54751e0991140529859096c1f3fc6938719#diff-949e07df8a5fc0851347cd6e6a971a84 .

**Pull request merged**